### DISCUSSION TITLE
/posts/how-to-setup-alacritty-terminal

### DISCUSSION THREAD

**Discussion by giscus** [josean-dev/sveltekit-blog#12] [General]
# /posts/how-to-setup-alacritty-terminal

This is my step by step guide on how to setup Alacritty for an amazing minimal & fast terminal setup for macOs

https://www.josean.com/posts/how-to-setup-alacritty-terminal

**Comment by sethhorsley** [comment]
Great videos! I was curious what you use to switch applications or spaces? I programmed alt/option + 1 and alt/option + 2 .. to go to that number space or create it. Just wondering what your workflow is.

I noticed your alt key on your keyboard is on the far bottom right, is that an easy key placement for common use? You are using it everywhere with yabai and skhd so I am super curious about how your workflow works.

Basically:
1. What shortcuts do you use to switch spaces and switch applications?
2. Is your alt key keyboard location convenient for common use on your custom keyboard?
3. Could you give us more tips on how you use your computer?

**Reply by josean-dev** [reply]
My alt key is my left thumb. Its a mod tap key so tapping is enter and holding is alt/option. To switch spaces I use ctrl + # of space.

**Comment by silicooon** [comment]
I am having trouble only in the last step with control a and shift I
It doesn't do anything and alacritty seems confused... 


Like that.. I am typing correctly but gibberish stuff appears.. 
![tmux](https://i.imgur.com/a9kzvtC.png)

**Reply by josean-dev** [reply]
Did you download the `.tmux.conf` file with the command and start tmux? If you already had tmux running before downloading the `.tmux.conf` file, then in tmux run `tmux source ~/.tmux.conf`. Not sure if this is the issue. Also make sure you first do `Ctrl+a`, pressing down both keys together and then let go, then right after that do `Shift+i` pressing down both keys together and then let go.

**Reply by silicooon** [reply]
Nothing happens, I have tried source before as well.. it seems that when I am on tmux I "cannot" type correctly or something..

**Comment by alexb-dev** [comment]
the same

**Comment by EvanChan0** [comment]
same

**Reply by impersin** [reply]
use this set -g default-terminal "screen-256color"

**Comment by silicooon** [comment]
its only when i am trying to type something on tmux, like there's a bug or something like it.
I have removed it from brew, deleted the config. Installed it again and then tried it and it worked.
Then I put back the config and again the typing on tmux goes crazy.. if i type exit and enter, it understands it but in my shell i see something like eeeexxit. :/

Any idea? Thank so much in advance!

**Reply by josean-dev** [reply]
I'm sorry this is happening. I'm honestly not sure what it might be, as it works properly in my own testing/setup. I would recommend you add lines to the config one by one until you find what might be causing the issue. To reload the config when you modify it you can do `tmux source ~/.tmux.conf`.

**Comment by flavono123** [comment]
thank you for sharing yours and also recommend my one for clickable urls in alacritty:
https://github.com/flavono123/dotfiles/commit/c25fb39e175fd08428082e322eb99e3e6c3e03d7

**Comment by isola513i** [comment]
How to fullscreen ?

**Comment by noahra** [comment]
```bash
git clone https://github.com/tmux-plugins/tpm ~/.tmux/plugins/tpm
```

Try doing this before `ctrl+a` + `shift+I`

**Reply by ross-mclain-br** [reply]
Yes please add this as a step to make sure tpm is ready before installing plugins. When I first opened tmux, I hit ctrl+a + shift+I and nothing happens until after I googled how to install plugins and found that tpm was missing. Otherwise awesome tutorial that has really helped my daily flow of work!

**Comment by going2river** [comment]
Hello!  I also had problems with Tmux working.  Please tell me how to roll back everything done and delete everything?

**Comment by bajra-manandhar17** [comment]
If the tmux is not reading the `ctrl a, shift i` bind, try to reload the config with this line
```
tmux source-file ~/.tmux.conf
```

**Comment by code-lwh** [comment]
Hey josean, i wanted to email you to show you something i made using a document database that was inspired by your course app which is in the making, please shoot me an email at **lwhpers@gmail.com** because i could not find your email. I think it is pretty amazing and i just wanted to tell you about it to learn more

**Reply by josean-dev** [reply]
Hey! Awesome! You can send me an email at youtube@josean.com. Looking forward to hearing more!

**Comment by swastkk** [comment]
How to handle this issue??
https://github.com/alacritty/alacritty/issues/607
Cant have multiple desktops for alacritty!

**Reply by swastkk** [reply]
@josean-dev ??

**Reply by swastkk** [reply]
I sorted out this issue, this issue was due to basic settings by macOS itself, and can be solved by 
Double pressing the Alacritty icon in the dock, then go to `Options` and then `Assign to` and then Choose None,
it will not be moving if we switch the desktops, it was quite annoying.

**Comment by callmesukhi** [comment]
@josean-dev Alacritty setup broke my VS Code terminal, unfortunately. Do you know how can I fix this? 

Link to the image: https://imgur.com/a/jrsElH0

**Reply by ppoxx** [reply]
Looks to me that the terminal in VS code is not using a Nerd Font thus cannot display the icons etc...

**Comment by Dkerr2061** [comment]
Everything was working with Alacritty and then I decided to change the theme and for some reason, my cd command doesn't work anymore. This is the error that I get: 
cd ~
zsh: command not found: zsource

any thought's on how I can fix this?

**Reply by yddod-ayajima** [reply]
what is your alias looks like for cd in .zshrc?

**Reply by Dkerr2061** [reply]
I fixed it, I just had to use the chdir command to access the .zshrc file. I guess that I forgot to save the changes before closing it out on VSCode.

**Comment by yddod-ayajima** [comment]
@josean-dev how do you get the filename completion inside nvim when you are typing the theme path and name?

**Comment by cug** [comment]
After seeing this blog post, I thought I'd try alacritty and tmux on my Mac. What a disaster ...

It took me forever to get everything to the point where automatic theme switching (dark/light switching when the system appearance was changed either automatically or manaully in System Settings)  actually worked in alacritty, tmux, and nvim. And the only way I found to do that was really ugly, messing with config files via sed etc.

What a pain for something that mostly "just works" with iTerm2. The only thing there is that nvim doesn't switch automatically. But with nvim sessions that's really not that much of a problem.

I'm going to happily stay with iTerm2 and without tmux.

**Reply by cug** [reply]
Oh, and tmux-resurrect kept resurrecting some old sessions that I had killed multiple times. Trying tmux attach reported no session, starting tmux resulted in some old zombie session to be resurrected ... had to clean this out and disabled the plugin.

**Comment by jg-edu** [comment]
Hi everyone - if you're running into the issue where you launch tmux and then you get a bunch of gibberish as you're typing or double typing - try this link: https://gist.github.com/bbqtd/a4ac060d6f6b9ea6fe3aabe735aa9d95 - it solved my issue.

**Reply by fuadmhnr** [reply]
Thank you you're the man

**Comment by abelgroot** [comment]
background transparency is not working in fullscreen mode

**Comment by josegarnier** [comment]
Hi ! Thank you for the blog and video. Is it normal if the "apple terminal" is also changed ? Because for me it is and it isn't displayed correctly (we can't display icons, the theme is not applied etc...). How to put the original terminal as it was while keeping the changes for Alacrity ? 

Also in Alacrity with eza the accentuations aren't well displayed (for example é,è,...) and also i can't type '~' (with alt+n)...

Finally, in both terminals, when i change the size of the windows something weird happens and the display of the path multiply itself making it kinda non-readable.

By the way i'm on an Intel Mac with Monterey and I'm french and a really beginer in dev. (Sorry if i didn't putted pictures, i don't know how to ... :/ )

**Comment by neil-mando** [comment]
Guys if you can't install the plugins by pressing CTRL-A followed by Shift-I
First, you need to  `git clone https://github.com/tmux-plugins/tpm ~/.tmux/plugins/tpm`
then `tmux source ~/.tmux.conf`

After that go ahead enter tmux and now you can install all plugins from @josean-dev tutorial

**Reply by nahugonza11** [reply]
This was really helpful, it worked for me.

**Comment by horyekhunley** [comment]
I'm using linux, will the commands work for me. Also getting this error when i run "tmux source-file ~/.tmux.conf": 

'~/.tmux/plugins/tpm/tpm' returned 1

**Comment by robinmahz** [comment]
the delete is not working properly. whenever I press the delete key instead of deleting a character, it appends extra spaces like pressing tab.

**Comment by rokibulroni** [comment]
One single command, and you are all set up,

/bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)" && echo 'eval "$(/opt/homebrew/bin/brew shellenv)"' >> ~/.zprofile && source ~/.zprofile && brew install --cask alacritty && brew install git font-meslo-lg-nerd-font powerlevel10k eza zoxide tmux && mkdir -p ~/.config/alacritty && touch ~/.config/alacritty/alacritty.toml && echo "source $(brew --prefix)/share/powerlevel10k/powerlevel10k.zsh-theme" >> ~/.zshrc && source ~/.zshrc && git clone https://github.com/alacritty/alacritty-theme ~/.config/alacritty/themes && git clone https://github.com/tmux-plugins/tpm ~/.tmux/plugins/tpm && curl https://raw.githubusercontent.com/josean-dev/dev-environment-files/main/.tmux.conf -o ~/.tmux.conf && tmux start && tmux source ~/.tmux.conf

**Comment by MoJavadSh** [comment]
Hey there
anyone knows how to restore the terminal to the default setting?

**Comment by abbaty48** [comment]
Amazen, Awesome and very usefull article.
I have configure it with my Arch-Craft
